api/net/ip4/cidr.hpp

```diff
@@ -0,0 +1,65 @@
+// This file is a part of the IncludeOS unikernel - www.includeos.org
+//
+// Copyright 2015-2017 Oslo and Akershus University College of Applied Sciences
+// and Alfred Bratterud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NET_IP4_CIDR_HPP
+#define NET_IP4_CIDR_HPP
+
+#include "addr.hpp"
+
+namespace net {
+namespace ip4 {
+
+class Cidr {
+public:
+  Cidr(uint8_t p1, uint8_t p2, uint8_t p3, uint8_t p4, uint8_t mask)
```

Comment:
* Can be probably be `constexpr` (if members are made `const`, which they can since the class is immutable), and `noexcept`.
* Not sure if any need for Cidr(Addr&& addr, uint8_t mask) (maybe just wait with it?)
* Some short explanation of how to use the constructor would be nice!